src/main/java/duke/logic/conversations/Conversation.java

```diff
@@ -44,33 +46,85 @@ public String getPrompt() {
     /**
      * Gets result of prompt.
      *
-     * @return result The result
+     * @return result The result.
      */
     public String getResult() {
         assert (result != null) : "result should not be null";
         return result;
     }
 
     /**
-     * Checks and sets input if is int.
-     * @param input The input
-     * @return String The input if is int
+     * Checks if input is int.
+     *
+     * @param input The input.
+     * @return If the input is int.
      */
     protected Boolean isIntInput(String input) {
```

Comment:
We should extract all of these check conditions into one class each

Reply:
Extracted into ConversationUtil.java.